### PR TITLE
chore: Upgrade to Go 1.21, several AWS dependencies

### PR DESCRIPTION
**Pull request by bhavanki**
I filed this as an alternative to five other dependabot PRs, and to investigate why they were failing. Once this one is merged, those other PRs will become obsolete.

The root problem discovered here is that the AWS SDK just increased its minimum Go version to 1.21. Chamber may do the same, since it only supports the most recent _two_ Go versions. Although 1.21 is now the third most recent, we'll leave it in since it can at least build.

https://github.com/segmentio/chamber/pull/559 also groups future aws-sdk-go-v2 upgrades into single PRs, for easier review and more efficient use of GitHub resources.

### CODE REVIEW

**Comment by bhavanki** [comment]
Ah, the AWS SDK now requires Go 1.21. https://github.com/aws/aws-sdk-go-v2/releases/tag/release-2024-08-15

File: go.mod [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
 require (
 	github.com/alessio/shellescape v1.4.2
-	github.com/aws/aws-sdk-go-v2 v1.30.3
-	github.com/aws/aws-sdk-go-v2/config v1.27.27
-	github.com/aws/aws-sdk-go-v2/feature/ec2/imds v1.16.11
+	github.com/aws/aws-sdk-go-v2 v1.30.4
+	github.com/aws/aws-sdk-go-v2/config v1.27.28
+	github.com/aws/aws-sdk-go-v2/feature/ec2/imds v1.16.12
 	github.com/aws/aws-sdk-go-v2/service/s3 v1.57.1
 	github.com/aws/aws-sdk-go-v2/service/secretsmanager v1.32.1
```
Comment (alecjacobs5401) [inline go.mod:13]: Post merge of https://github.com/segmentio/chamber/pull/559, https://github.com/segmentio/chamber/pull/560 was opened which also shows bumps to these two packages. Should we bump these as well to keep everything aligned?

Comment (bhavanki) [reply go.mod:13]: Oh yesh, adding them

**Review by alecjacobs5401 — DISMISSED** [review]

File: go.mod [hunk 2]
```diff
@@ -1,17 +1,19 @@
 module github.com/segmentio/chamber/v3
 
-go 1.20
+go 1.21
+
+toolchain go1.21.6
```
[the remark this answers is not in the document]
Comment (bhavanki) [reply go.mod]: Turns out we don't need the `toolchain` line. `go mod tidy` added it, I suppose based on what I have locally, but `go get toolchain@none`([ref](https://tip.golang.org/doc/toolchain#get)) removed it, and everything still seems OK.

[the remark this answers is not in the document]
Comment (bhavanki) [reply go.mod]: Oh, as for the Dockerfile, dependabot is already on it: https://github.com/segmentio/chamber/pull/557/files

**Pull request merged**